### PR TITLE
Store sailbot gps

### PR DESCRIPTION
**Pull request by JoshuaJSam**
To address: https://github.com/UBCSailbot/local-pathfinding/issues/129

Created an independent ros node that subscribes to the sailbot GPS. Every second stores the sailbot GPS. Then at the end, plots the full path that the boat took (see how jagged or uneven it is) and total displacement from starting point.

### CODE REVIEW

**Comment by JoshuaJSam** [comment]
@tylerlum I tried using some existing code from log_closest_obstacle.py which creates a png plot at the end. The code using "ABS_PATH_TO_OUTPUT_SAILBOT_GPS_DISPLACEMENT" . I am getting an error though outputing the plot

File: python/store_sailbot_gps.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        plt.clf()
+
+    
+if __name__ == '__main__':
+    rospy.init_node('SailbotGPSData')
+    myClass = SailbotGPSData()
+    while not rospy.is_shutdown():
+        myClass.StoreSailbotGPS()
+    
+    rospy.loginfo("hi")
```
Comment (tylerlum) [inline python/store_sailbot_gps.py:77]: Change this to "Saving to {}".format(ABS_PATH_TO_OUTPUT_SAILBOT_GPS_DISPLACEMENT)

**Review by tylerlum — CHANGES_REQUESTED** [review]
Add this to the `all.launch` file like log_closest_obstacle.py.

Also, you may need to run `chmod +x python/store_sailbot_gps.py` to ensure that this can be executable

**Comment by JoshuaJSam** [comment]
@tylerlum Its finished now :)

**Comment by JoshuaJSam** [comment]
@patrick-5546 If you would like to review this PR its functionality is pretty straightforward. If it is working properly then it will create a numpy plot of the displacement in a local-pathfinding folder where the other plots are stored as well. It should print out the file path after you CTRL+C and end the ros simulation.

File: python/store_sailbot_gps.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        plt.grid()
+        plt.xlabel('Longitude')
+        plt.ylabel('Latitude')
+        plt.draw()
+        plt.pause(0.0001)
+        plt.clf()
+        '''
+    
+if __name__ == '__main__':
+    rospy.init_node('SailbotGPSData')
```
Comment (patrick-5546) [inline python/store_sailbot_gps.py:73]: For clarity, this line can go in `__init__()` of `SailbotGPSData`

File: python/store_sailbot_gps.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        #Update the values for Lat and Lon
+        self.lat = data.lat
+        self.lon = data.lon
+
+    def Find_Distance(self):
+        return distance((self.latInitial, self.lonInitial), (self.lat, self.lon)).km
+
+    def StoreSailbotGPS(self):
+        #sleep for 1 second, so that lat and lon values are updated every second
+        rospy.sleep(UPDATE_TIME_SECONDS)
```
Comment (patrick-5546) [inline python/store_sailbot_gps.py:51]: For clarity, this can go inside `while not rospy.is._shutdown()`

**Review by patrick-5546 — COMMENTED** [review]
Looks good to me. You can see where and what your lint errors are by clicking details.

**Review by tylerlum — APPROVED** [review]
Awesome work! Feel free to merge

**Pull request merged**